specdb/build/utils.py

```diff
@@ -648,6 +663,119 @@ def set_resolution(head, instr=None):
     else:
         raise IOError("Not read for this instrument")
 
+def parse_header(parse_head:dict, head, mdict:dict=None, 
+                 plist:list=None, count=0):
+    """ Method to parse the items in the header
+
+    Args:
+        parse_head : dict
+            Parse header for meta info with this dict
+        head (fits.Header):  FITS header to parse 
+        mdict (_type_, optional): _description_. Defaults to None.
+        mdict (dict, optional):
+            Input meta data in dict form e.g.  mdict=dict(INSTR='ESI')
+        plist (dict, optional): Dictionary that holds 
+            the lists of parameters.  If input, it will be
+            modified in place
+        count (int, optional): Running count
+
+    Raises:
+        ValueError: _description_
+
+    Returns:
+        dict: plist,  dict that holds the list of parameters
+    """
+
+    Rdicts = defs.get_res_dicts()
+
+    if plist is None:
+        plist = {}
+        for key in parse_head.keys():
+            plist[key] = []
+
+    for key,item in parse_head.items():
+        # R
+        if key == 'R':
+            if parse_head[key] is True:
+                try:
+                    plist[key].append(set_resolution(head))
+                except ValueError:
+                    if mdict is not None:
+                        try:
+                            plist[key].append(mdict['R'])
+                        except KeyError:
+                            print("Key error!")
+                            pdb.set_trace()
+                    else:
+                        print('Bad inputs to set_resolution()')
+                        pdb.set_trace()
+                        plist[key].append(0.)
+            else:
+                raise ValueError("Set something else for R")
+        elif key == 'DATE-OBS':
+            if 'MJD' in item:
+                tval = Time(head[item], format='mjd')
+                tval.format = 'iso'
+            else:
+                if ':' in head[item]:
+                    thead = head[item].replace(':','-')
+                elif '/' in head[item]:
+                    thead = head[item].replace('/','-')
+                else:
+                    thead = head[item]
+                tval = Time(thead, format='iso')
+            tval.out_submfit = 'date'
+            plist[key].append(tval.iso)
+        else:
+            plist[key].append(head[item])
+
+    # INSTRUMENT SPECIFIC
+    try:
+        instr = head['INSTRUME']
+    except KeyError:
+        instr = 'none'
+    # LRIS
+    if 'LRIS' in instr:
+        # Init
+        for skey in ['DISPERSER', 'INSTR', 'R']:
+            if skey not in plist.keys():
+                plist[skey] = []
+        # Figure out detector
+        try:
+            det = head['DETECTOR']
+        except KeyError:
+            if 'BLUE' in instr:
+                det = 'LRIS-B'
+            else:
+                if head['OUTFILE'] == 'lred':
```

Comment:
Is this for a 1D spec file? I don't see a keyword named `OUTFILE` in recent LRIS-r raw frames.

Reply:
must be for older files (before the new chip)